tests/tests.c

```diff
@@ -1431,10 +1436,44 @@ bool sirtest_os_log(void) {
 #endif
 }
 
+char *sirtest_get_wineversion(void) {
+#if defined(__WIN__)
+# if defined(_MSC_VER) && !defined(__clang__)
+#  pragma warning(push)
+#  pragma warning(disable : 4152)
+# endif
+    static const char* (CDECL* _p_wine_get_version)(void);
+    HMODULE _h_ntdll;
+
+    _h_ntdll = GetModuleHandle("ntdll.dll");
+    if (_h_ntdll != NULL) {
+# if defined(__GNUC__)
+#  pragma GCC diagnostic push
+#  pragma GCC diagnostic ignored "-Wpedantic"
+# endif
+        _p_wine_get_version = (void *)GetProcAddress(_h_ntdll, "wine_get_version");
```

Comment:
Is this the line that triggers warning 4152? Check out the example here. I believe if you do it like this, you won't get a warning (use STDCALL instead of CDECL, and typedef that function signature, then cast to that type not void*)

https://learn.microsoft.com/en-us/cpp/build/getprocaddress?view=msvc-170

Reply:
I'll check it out shortly, but it needs to work on more than just MSVC ... so pretty sure the warning still happens elsewhere, let me see ...

Reply:
@aremmell I'm stuck downstairs for a bit away from a real keyboard so if you want to try it and we'll see what CI thinks, feel free, unless I get to it first